core/src/main/java/io/fabric8/maven/core/openshift/BuildService.java

```diff
@@ -138,4 +137,25 @@ public void onClose(KubernetesClientException cause) {
             };
     }
 
+    private void logBuildBuildFailedDetails(OpenShiftClient client, String buildName) {
+        try {
+            BuildConfig build = client.buildConfigs().withName(buildName).get();
+            ObjectReference ref = build.getSpec().getStrategy().getSourceStrategy().getFrom();
+            String kind = ref.getKind();
+            String name = ref.getName();
+
+            if ("DockerImage".equals(kind)) {
+                log.error("Please, ensure that the Docker image '%s' exists and is accessible by OpenShift", name);
+            } else if ("ImageStreamTag".equals(kind)) {
+                String namespace = ref.getNamespace() != null ? ref.getNamespace() : "";
```

Comment:
nit: I think this block would be more readable like:

```java
String namespace = ref.getNamespace();
String namespaceInfo = "current";
String namespaceParams = "";
if (!namespace.isEmpty()) {
  String namespaceInfo = "'" + namespace + "'";
  String namespaceParams = " -n " + namespace;
}
```

but not really bothered if you don't agree.